module/VuFind/src/VuFind/Config/Feature/ConfigSettingPropertiesTrait.php

```diff
@@ -0,0 +1,252 @@
+    ): array {
+        return $this->requiredSettings[$contextKey];
+    }
+
+    /**
+     * Is the setting required?
+     *
+     * The optional context and context key parameters are used to evaluate if a
+     * conditionally required setting is required. If context is omitted returns
+     * true for both required and conditionally required settings.
+     *
+     * @param string $setting    Setting
+     * @param array  $context    Settings to be used in evaluation (optional)
+     * @param string $contextKey Key identifying the context (optional)
+     *
+     * @return bool
+     */
+    public function isRequiredSetting(
+        string $setting,
+        array $context = [],
+        string $contextKey = self::DEFAULT_CONTEXT
+    ): bool {
+        // Default implementation does not handle conditional requirements.
+        return in_array($setting, $this->getRequiredSettings($contextKey));
+    }
+
+    /**
+     * Return settings that may be localized.
+     *
+     * @param string $contextKey Key identifying the context (optional)
+     *
+     * @return array
+     */
+    public function getLocalizableSettings(
+        string $contextKey = self::DEFAULT_CONTEXT
+    ): array {
+        return $this->localizableSettings[$contextKey];
+    }
+
+    /**
+     * Is the setting localizable?
+     *
+     * @param string $setting    Setting
+     * @param string $contextKey Key identifying the context (optional)
+     *
+     * @return bool
+     */
+    public function isLocalizableSetting(
+        string $setting,
+        string $contextKey = self::DEFAULT_CONTEXT
+    ): bool {
+        return in_array($setting, $this->getLocalizableSettings($contextKey));
+    }
+
+    /**
+     * Validate settings.
+     *
+     * @param array  $settings   Settings
+     * @param string $contextKey Key identifying the context (optional)
+     *
+     * @return array
+     * @throws BadConfig
+     */
+    public function validateSettings(
+        array $settings,
+        string $contextKey = self::DEFAULT_CONTEXT
+    ): array {
+        foreach ($this->getRequiredSettings($contextKey) as $requiredSetting) {
+            if (
+                !isset($settings[$requiredSetting])
+                && $this->isRequiredSetting($requiredSetting, $settings, $contextKey)
+            ) {
+                throw new BadConfig(
+                    'Missing required setting: ' . $requiredSetting
+                );
+            }
+        }
+        return $settings;
+    }
+
+    /**
+     * Localize the settings if possible.
+     *
+     * @param array  $settings        Settings to localize
+     * @param string $userLocale      User locale
+     * @param array  $fallbackLocales Fallback locale(s) (optional)
+     * @param string $contextKey      Key identifying the context (optional)
+     * @param bool   $useFirst        Use first array item if item matching
+     *                                locale(s) was not found (optional)
```

Comment:
I don't think this needs to be wrapped:

```suggestion
     * @param bool   $useFirst        Use first array item if item matching locale(s) was not found (optional)
```